tests/test_round.py

```diff
@@ -871,5 +873,207 @@ class FakeModule:
+
+    def test_round_19_process_optim_aux_var_without_base_optimizer(self):
+        """Test that 'process_optim_aux_var' documents missing BaseOptimizer."""
+        # Set up a Round with fake aux_vars, but no BaseOptimizer.
+        rnd = Round()
+        setattr(rnd, "_optim_aux_var", {"module": {"key": "val"}})
+        rnd.training_plan = create_autospec(BaseTrainingPlan, instance=True)
+        rnd.training_plan.optimizer.return_value = None
+        # Call the tested method, verifying that it returns an error.
+        msg = rnd.process_optim_aux_var()
+        self.assertTrue("TrainingPlan does not hold a BaseOptimizer" in msg)
+
+    def test_round_20_process_optim_aux_var_without_optimizer(self):
+        """Test that 'process_optim_aux_var' documents missing Optimizer."""
+        # Set up a Round with aux vars, but a non-Optimizer optimizer.
+        rnd = Round()
+        setattr(rnd, "_optim_aux_var", {"module": {"key": "val"}})
+        mock_b_opt = create_autospec(BaseOptimizer, instance=True)
+        mock_b_opt.optimizer = MagicMock()  # not a declearn-based Optimizer
+        rnd.training_plan = create_autospec(BaseTrainingPlan, instance=True)
+        rnd.training_plan.optimizer.return_value = mock_b_opt
+        # Call the tested method, verifying that it returns an error.
+        msg = rnd.process_optim_aux_var()
+        self.assertTrue("does not manage a compatible Optimizer" in msg)
+
+    def test_round_21_process_optim_aux_var_with_optimizer_error(self):
+        """Test that 'process_optim_aux_var' documents 'Optimizer.set_aux' error."""
+        # Set up a Round with fake pre-downloaded aux vars.
+        rnd = Round()
+        fake_aux_var = {"module": {"key": "val"}}
+        setattr(rnd, "_optim_aux_var", fake_aux_var)
+        # Set up a mock BaseOptimizer with an attached failing Optimizer.
+        mock_optim = create_autospec(Optimizer, instance=True)
+        fake_error = "fake FedbiomedOptimizerError on 'set_aux' call"
+        mock_optim.set_aux.side_effect = FedbiomedOptimizerError(fake_error)
+        mock_b_opt = create_autospec(BaseOptimizer, instance=True)
+        mock_b_opt.optimizer = mock_optim
+        # Attach the former to the Round's mock TrainingPlan.
+        rnd.training_plan = create_autospec(BaseTrainingPlan, instance=True)
+        rnd.training_plan.optimizer.return_value = mock_b_opt
+        # Call the tested method, verifying that it returns an error.
+        msg = rnd.process_optim_aux_var()
+        self.assertTrue(fake_error in msg)
+        mock_optim.set_aux.assert_called_once_with(fake_aux_var)
+
+    def test_round_22_collect_optim_aux_var(self):
+        """Test that 'collect_optim_aux_var' works properly with an Optimizer."""
+        # Set up a Round with an attached mock Optimizer.
+        rnd = Round()
+        mock_optim = create_autospec(Optimizer, instance=True)
+        mock_b_opt = create_autospec(BaseOptimizer, instance=True)
```

Comment:
I do not think so. I mean, in practice, in the current state of things, it would be a `DecLearnOptimizer`, but here we are using mock objects, and by the signatures of the tested objects, it is enough to pass a `BaseOptimizer`-like mock, the `.optimizer` of which returns an `Optimizer`.

Reply:
I see, but such auxiliary variable exchange is specific to `DecLearnOptimizer` , so for me it would make more sense to use it instead of doing something more generic. But since the tests are passing, i am ok to leave it as it is